tpl-implementations/include/experimental/__p1673_bits/kokkos-kernels/blas3_matrix_product_kk.hpp

```diff
@@ -761,6 +773,50 @@ void triangular_matrix_left_product(
   }
 }
 
+// Overwriting triangular matrix-matrix left product
+// performs in-place BLAS xTRMM: C = A x C
+
+MDSPAN_TEMPLATE_REQUIRES(class ExecSpace,
+    class ElementType_A,
+    std::experimental::extents<>::size_type numRows_A,
+    std::experimental::extents<>::size_type numCols_A,
+    class Layout_A,
+    class Triangle,
+    class DiagonalStorage,
+    class ElementType_C,
+    std::experimental::extents<>::size_type numRows_C,
+    std::experimental::extents<>::size_type numCols_C,
+    class Layout_C,
+    /* requires */ (Impl::is_unique_layout_v<Layout_C, numRows_C, numCols_C>
+        and (Impl::is_unique_layout_v<Layout_A, numRows_A, numCols_A>
+        or Impl::is_layout_blas_packed_v<Layout_A>)))
+void triangular_matrix_left_product(
+  kokkos_exec<ExecSpace>&& exec,
+  std::experimental::mdspan<ElementType_A, std::experimental::extents<numRows_A, numCols_A>, Layout_A,
+    std::experimental::default_accessor<ElementType_A>> A,
+  Triangle t,
+  DiagonalStorage d,
+  std::experimental::mdspan<ElementType_C, std::experimental::extents<numRows_C, numCols_C>, Layout_C,
+    std::experimental::default_accessor<ElementType_C>> C)
+{
+  matproduct_impl::check_left_product(A, t, C, C, "triangular_matrix_left_product");
+
+  Impl::signal_kokkos_impl_called("overwriting_inplace_triangular_matrix_left_product_kokkos");
+
+  // convert mdspans to views
+  const auto A_view = Impl::mdspan_to_view(A);
+  auto C_view = Impl::mdspan_to_view(C);
+
+  // implicit diagonal is not supported by KK implementation of TRSM
+  // and the execution space is ignored
+  if constexpr (std::is_same_v<DiagonalStorage,
+                std::experimental::linalg::implicit_unit_diagonal_t>) {
+    matproduct_impl::trmm_left(ExecSpace(), t, d, A_view, C_view, C_view);
```

Comment:
Please do consider storing the actual execution space instance in `kokkos_exec` at some point.